tests/jquery.flot.composeImages.Test.js

```diff
@@ -0,0 +1,698 @@
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(50, 130, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 0, 255, 255)).toBe(true);
+
+            //other points on circles should match the required colors, because of configured diameters
+            pixelData = destinationCanvas.getContext('2d').getImageData(220, 17, 1, 1).data;
+            expect(matchPixelColor(pixelData, 255, 0, 0, 255)).toBe(true);
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(190, 114, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 255, 0, 255)).toBe(true);
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(80, 149, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 0, 255, 255)).toBe(true);
+            expect(pixelData).toMatchPixelColor([0, 0, 255, 255]);
+
+            //verify a pixel from the circle border, if it comes from a black line (almost black, because of antialiasing), as described in svg style
+            pixelData = destinationCanvas.getContext('2d').getImageData(79, 102, 1, 1).data;
+            expect(matchPixelColorWithError(pixelData, 0, 0, 0, 255, 15)).toBe(true);
+
+            done();
+        }, null);
+    });
+
+
+    function drawSomeLinesOnCanvas(canvas) {
+        var ctx = canvas.getContext('2d');
+        ctx.beginPath();
+        ctx.moveTo(0, 0);
+        ctx.lineTo(19, 19);
+        ctx.moveTo(3, 18);
+        ctx.lineTo(17, 5);
+        ctx.stroke();
+    }
+
+    function drawARectangleOnCanvas(canvas, color) {
+        var ctx = canvas.getContext('2d');
+        ctx.rect(0, 0, 20, 20);
+        ctx.fillStyle = color;
+        ctx.fill();
+    }
+
+    function matchPixelColor(pixelData, r, g, b, a) {
+        return (pixelData[0] === r) && (pixelData[1] === g) && (pixelData[2] === b) && (pixelData[3] === a);
+    }
+
+    function matchPixelColorWithError(pixelData, r, g, b, a, err) {
+        return (Math.abs(pixelData[0] - r) <= err) && (Math.abs(pixelData[1] - g) <= err) && (Math.abs(pixelData[2] - b) <= err) && (Math.abs(pixelData[3] - a) <= err);
+    }
+
+    function matchPixelDataArrays(pixelData1, pixelData2) {
+        var sameValue = true;
+        if (pixelData1.length !== pixelData2.length) {
+            sameValue = false;
+        } else {
+            for (var i = 0; i < pixelData1.length; i++) {
+                if (pixelData1[i] !== pixelData2[i]) {
+                    sameValue = false;
+                    break;
+                }
+            }
+        }
+        return sameValue;
+    }
+
+    //see https://jasmine.github.io/2.0/custom_matcher.html
+    var pixelMatchers = {
+        toMatchPixelColor: function(util, customEqualityTesters) {
+            return {
+                compare: function(actual, expected) {
+                    if (expected === undefined) {
+                        expected = [-1000, -1000, -999, -999]; //no color should match these values
+                    }
```

Comment:
There is a common JS pattern for this case:
```js
expected = expected || [-1000, -1000, -999, -999];
```
Not extremely useful in this particular case, but it makes the code much cleaner when the function has a lot of arguments.

Reply:
done